crates/secrets-types/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+use anyhow::ensure;
+use async_trait::async_trait;
+use nkeys::XKey;
+use serde::{Deserialize, Serialize};
+use wascap::jwt::{validate_token, CapabilityProvider, Component, Host};
+
+mod errors;
+pub use crate::errors::*;
+
+/// The key of a NATS header containing the wasmCloud host's public xkey used to encrypt a secret request.
+/// It is also used to encrypt the response so that only the requestor can decrypt it.
+pub const WASMCLOUD_HOST_XKEY: &str = "WasmCloud-Host-Xkey";
+pub const RESPONSE_XKEY: &str = "Server-Response-Xkey";
+
+/// The request context for retrieving a secret
+#[derive(Serialize, Deserialize, Default)]
+pub struct Context {
+    /// The component or provider's signed JWT.
+    pub entity_jwt: String,
```

Comment:
Perhaps this is covered somewhere, but after taking another look at this, I'm wondering what's the reasoning to prefer `entity_jwt: String` over `component_jwt: Option<String>` and `provider_jwt: Option<String>`?

Reply:
I prefer not having separate jwt fields for components and providers. You should always have a JWT and it doesn't matter much which thing it is coming from

Reply:
I agree with @thomastaylor312 which is why I implemented it this way. The JWT should be present regardless of the type, so distinguishing between them doesn't matter.